presto-orc/src/main/java/com/facebook/presto/orc/AbstractOrcRecordReader.java

```diff
@@ -548,6 +551,20 @@ private void advanceToNextStripe()
         }
     }
 
+    private List<Slice> getDecryptionKeyMetadata(int currentStripe)
+    {
+        // if this stripe has encryption keys, then those are used
+        // otherwise look at nearest prior stripe that specifies encryption keys
+        // if the first stripe has no encryption information, then there are no encrypted columns;
+        for (int i = currentStripe; i >= 0; i--) {
+            if (!stripes.get(i).getKeyMetadata().isEmpty()) {
+                return stripes.get(i).getKeyMetadata();
```

Comment:
do we have check somewhere that checks key count matches number of encryption groups

Reply:
@rschlussel What do you think about @zzhao0 's question here?

Reply:
I added checks for that when we read the footer in OrcReader (validateEncryption()